pkg/kratos/service.go

```diff
@@ -180,7 +185,7 @@ func (s *Service) GetRecoveryFlow(ctx context.Context, id string, cookies []*htt
 	return flow, resp.Cookies(), nil
 }
 
-func (s *Service) GetSettingsFlow(ctx context.Context, id string, cookies []*http.Cookie) (*kClient.SettingsFlow, []*http.Cookie, error) {
+func (s *Service) GetSettingsFlow(ctx context.Context, id string, cookies []*http.Cookie) (*kClient.SettingsFlow, []*http.Cookie, *BrowserLocationChangeRequired, error) {
```

Comment:
this methods seems to be doing too much

we have 4 returning parameters which in a lot of cases are `nil` 

would be worth refactoring that and in case duplicate logic or move some of the discriminating logic one layer up into the handler